### ISSUE TITLE
New branching concept and documentation

### ISSUE THREAD

**Issue by thomaszbz** [mblaschke/TYPO3-metaseo#451]
**MetaSEO version**: ~3

Now that we support MetaSEO versions 2.x and 3.x in parallel, we need a new branching concept.

## Structure

* **Feature** branches have prefix `feature-`. Please note that feature branches could be based upon an older version of metaseo for two reasons:

    * not up-to date to the master branch (current `master` needs to be merged into it)
    * intentionally => it's a backport

* `master` branch which is our new `default` branch. Replaces the old `develop` branch.

* **Hotfix** branches which contain commits for minor releases of already released major versions. They have the same name as the branch name of the release, suffixed with `-hotfixes`. "Hot" means: ready to be released.

* **Released versions**. The HEAD of such a branch is always a version-tagged commit. Prefixed with `v`, followed by the major version number (`x.y`), according to the version tag. That's one branch per major release (version `x.y`). To be used as an archive.

## Example of branch names

```
feature-abc
feature-xyz

master (patches for metaseo 3.1)

v3.0-hotfixes (unreleased patches for 3.0.1)
v2.1-hotfixes (unreleased patches for 2.1.1)
v2.0-hotfixes (unreleased patches for 2.0.5)

v3.0 (tagged and released 3.0.x versions)
v2.1 (tagged and released 2.1.x versions)
v2.0 (tagged and released 2.0.x versions)
v1.0 (tagged and released 1.0.x versions)
```

## Conventions

* `cherry-pick` is allowed from newer into older branches _only_ (backport).
* Never `cherry-pick` from old branches into new branches. Instead, do a `git merge`. Don't use `gitflow` here, respectively.
* **_Every_ commit that makes it into a hotfix branch for older versions than current** `master` **_must always and instantly_ be merged into  _each_ newer hotfix branch _and_ into master.** Resolve conflicts accordingly.
* If the change is supposed to intentionally _not_ make it into a newer version, then `git revert` it in a temporary local branch, just before merging.
* If done correctly, merging from an old hotfix branch into a newer hotfix branch or the master branch _never_ changes something.
